src/Commands/Utilities/RecycleBinUtility.cs

```diff
@@ -99,8 +105,44 @@ internal static List<RecycleBinItemCollection> GetRecycleBinItemCollection(Clien
                 }
             }
             while (items?.Count == 5000);
-            
+
             return recycleBinItems;
         }
+
+        internal static void RestoreRecycleBinItemInBulk(HttpClient httpClient, ClientContext ctx, string[] idsList)
+        {
+
+            Uri currentContextUri = new Uri(ctx.Url);
+            string apiCall = $"{currentContextUri}/_api/site/RecycleBin/RestoreByIds";
+
+            string idsString = string.Join("','", idsList); // Convert array to a comma-separated string  
+
+            try
+            {
+                string requestBody = $"{{'ids':['{idsString}']}}";
+                REST.RestHelper.Post(httpClient, apiCall,ctx, requestBody,"application/json", "application/json");
+            }
+            catch
+            {
+                //fall back logic
```

Comment:
Some verbose logging if this happens would be nice with the exception details so one can find out why it happens (i.e. too many items provided)